scenario/state.py

```diff
@@ -760,15 +862,16 @@ class State:
     stored_state: List["StoredState"] = dataclasses.field(default_factory=list)
     """Contents of a charm's stored state."""
 
-    # the current statuses. Will be cast to _EntitiyStatus in __post_init__
-    app_status: Union[StatusBase, _EntityStatus] = _EntityStatus("unknown")
+    # the current statuses.
+    app_status: _EntityStatus = UnknownStatus()
     """Status of the application."""
-    unit_status: Union[StatusBase, _EntityStatus] = _EntityStatus("unknown")
+    unit_status: _EntityStatus = UnknownStatus()
     """Status of the unit."""
     workload_version: str = ""
     """Workload version."""
 
     def __post_init__(self):
+        # Let people pass in the ops classes, and convert them to the appropriate Scenario classes.
```

Comment:
we need to document this very well.
I think this is a reasonable expectation:
```python
status = ops.ActiveStatus()
state = State(unit_status=status)
assert state.unit_status is status
```

so we need to find a way to make it clear why it doesn't work

Reply:
(nothing new by the way, it was already like this)

Reply:
> we need to document this very well.

+1

> I think this is a reasonable expectation:
> 
> ```python
> status = ops.ActiveStatus()
> state = State(unit_status=status)
> assert state.unit_status is status
> ```

Well, except that you would typically be asserting on the output state and so almost never should be using `is`, plus I don't think it's that reasonable to be using `is` with statuses anywhere - they aren't singletons and ops doesn't make any guarantees about the objects staying the same.

> so we need to find a way to make it clear why it doesn't work

👍